### PR TITLE
Fix triggering of telescopes in light emission package

### PR DESCRIPTION
**Pull request by tobiaskleiner**
Deleting the array trigger at two different positions in the code seems to be working for the layout and for the variable case. The same procedure for the axes offsets is not successful at the moment for the varying distance case.

### CODE REVIEW

**Comment by tobiaskleiner** [comment]
When running the script (with explicitly giving the DNUM_TELESCOPES=1), sim_telarray still sets up four telescopes.
This needs further investigation:
```

Start setting up telescope 1.
Preprocessor command: /workdir/sim_telarray/sim_telarray/bin/pfp -v -I. -DNUM_TELESCOPES=1 -DWITH_LOW_GAIN_CHANNEL -DMAX_GAINS=2 -DWITH_GSL_RNG -DSIMTEL_VERSION=1709915244 -DSIMTEL_RELEASE=20240780 -I../cfg/CTAiobuf_maximum=1000000000 -I. -I/workdir/sim_telarray/sim_telarray/cfg -I/workdir/sim_telarray/sim_telarray/cfg/common -I/workdir/sim_telarray/sim_telarray/cfg/hess -I/workdir/sim_telarray/sim_telarray/cfg/hess2 -I/workdir/sim_telarray/sim_telarray/cfg/hess3 -I/workdir/sim_telarray/sim_telarray/cfg/hess5000 -I/workdir/sim_telarray/sim_telarray/cfg/CTA -I/workdir/sim_telarray/sim_telarray/cfg/MAGIC -DMAX_GAINS=2 -DTELESCOPE=1 - < /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg
Applying 0 weak command-line configuration parameter(s).
Check additional configuration for telescope 1 from /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg.
Configuration for MSTN-04 - TELESCOPE 1
137 configuration lines were processed.
Applying 10 forced command-line configuration parameter(s).
Filled 8 metaparameters for global configuration.
Camera 1 by default requires 3 pixels per trigger group.
Reading setup for telescope 1 camera from file camera_CTA-MST-NectarCam_20191120_majority-3nn.dat
Pixel type 1 has a hexagonal (flat y) funnel onto a circular visible cathode.
Reading funnel angle efficiencies for telescope 1 from NectarCAM_lightguide_efficiency_POP_131019.dat
Table with 47 rows has been read from file NectarCAM_lightguide_efficiency_POP_131019.dat
Telescope 1 has 86 mirrors (from file mirror_CTA-100_1.20-86-0.04.dat).
Geometric area of 86 mirrors (inclination corrected): 106.26 m^2.
Diameter is approximately 13.775 m
Effective focal length given as 1644.5 for nominal 1600.0 cm.
Random reflections with 0.0068 deg.
Warning: trigger window does not cover full readout window.
(F)ADC data is simulated for a readout window of up to 105.0 ns,
trigger conditions in a 105.0 ns window starting 3.0 ns before (F)ADC data.
Camera filter 1-D transmission table applied for telescope 1.
Average night sky background of telescope 1: 0.218 p.e./ns (DC current: 1.40 uA)

Start setting up telescope 2.
Preprocessor command: /workdir/sim_telarray/sim_telarray/bin/pfp -v -I. -DNUM_TELESCOPES=1 -DWITH_LOW_GAIN_CHANNEL -DMAX_GAINS=2 -DWITH_GSL_RNG -DSIMTEL_VERSION=1709915244 -DSIMTEL_RELEASE=20240780 -I../cfg/CTAiobuf_maximum=1000000000 -I. -I/workdir/sim_telarray/sim_telarray/cfg -I/workdir/sim_telarray/sim_telarray/cfg/common -I/workdir/sim_telarray/sim_telarray/cfg/hess -I/workdir/sim_telarray/sim_telarray/cfg/hess2 -I/workdir/sim_telarray/sim_telarray/cfg/hess3 -I/workdir/sim_telarray/sim_telarray/cfg/hess5000 -I/workdir/sim_telarray/sim_telarray/cfg/CTA -I/workdir/sim_telarray/sim_telarray/cfg/MAGIC -DMAX_GAINS=2 -DTELESCOPE=2 - < /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg
Applying 0 weak command-line configuration parameter(s).
Check additional configuration for telescope 2 from /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg.
Configuration for MSTN-04 - TELESCOPE 2
137 configuration lines were processed.
Applying 10 forced command-line configuration parameter(s).
Filled 8 metaparameters for telescope ID 2 (itel=1).
Camera 2 by default requires 3 pixels per trigger group.
Reading setup for telescope 2 camera from file camera_CTA-MST-NectarCam_20191120_majority-3nn.dat
Pixel type 1 has a hexagonal (flat y) funnel onto a circular visible cathode.
Reading funnel angle efficiencies for telescope 2 from NectarCAM_lightguide_efficiency_POP_131019.dat
Table with 47 rows has been read from file NectarCAM_lightguide_efficiency_POP_131019.dat
Telescope 2 has 86 mirrors (from file mirror_CTA-100_1.20-86-0.04.dat).
Geometric area of 86 mirrors (inclination corrected): 106.26 m^2.
Diameter is approximately 13.775 m
Effective focal length given as 1644.5 for nominal 1600.0 cm.
Random reflections with 0.0068 deg.
Warning: trigger window does not cover full readout window.
(F)ADC data is simulated for a readout window of up to 105.0 ns,
trigger conditions in a 105.0 ns window starting 3.0 ns before (F)ADC data.
Camera filter 1-D transmission table applied for telescope 2.
Average night sky background of telescope 2: 0.218 p.e./ns (DC current: 1.40 uA)

Start setting up telescope 3.
Preprocessor command: /workdir/sim_telarray/sim_telarray/bin/pfp -v -I. -DNUM_TELESCOPES=1 -DWITH_LOW_GAIN_CHANNEL -DMAX_GAINS=2 -DWITH_GSL_RNG -DSIMTEL_VERSION=1709915244 -DSIMTEL_RELEASE=20240780 -I../cfg/CTAiobuf_maximum=1000000000 -I. -I/workdir/sim_telarray/sim_telarray/cfg -I/workdir/sim_telarray/sim_telarray/cfg/common -I/workdir/sim_telarray/sim_telarray/cfg/hess -I/workdir/sim_telarray/sim_telarray/cfg/hess2 -I/workdir/sim_telarray/sim_telarray/cfg/hess3 -I/workdir/sim_telarray/sim_telarray/cfg/hess5000 -I/workdir/sim_telarray/sim_telarray/cfg/CTA -I/workdir/sim_telarray/sim_telarray/cfg/MAGIC -DMAX_GAINS=2 -DTELESCOPE=3 - < /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg
Applying 0 weak command-line configuration parameter(s).
Check additional configuration for telescope 3 from /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg.
Configuration for MSTN-04 - TELESCOPE 3
137 configuration lines were processed.
Applying 10 forced command-line configuration parameter(s).
Filled 8 metaparameters for telescope ID 3 (itel=2).
Camera 3 by default requires 3 pixels per trigger group.
Reading setup for telescope 3 camera from file camera_CTA-MST-NectarCam_20191120_majority-3nn.dat
Pixel type 1 has a hexagonal (flat y) funnel onto a circular visible cathode.
Reading funnel angle efficiencies for telescope 3 from NectarCAM_lightguide_efficiency_POP_131019.dat
Table with 47 rows has been read from file NectarCAM_lightguide_efficiency_POP_131019.dat
Telescope 3 has 86 mirrors (from file mirror_CTA-100_1.20-86-0.04.dat).
Geometric area of 86 mirrors (inclination corrected): 106.26 m^2.
Diameter is approximately 13.775 m
Effective focal length given as 1644.5 for nominal 1600.0 cm.
Random reflections with 0.0068 deg.
Warning: trigger window does not cover full readout window.
(F)ADC data is simulated for a readout window of up to 105.0 ns,
trigger conditions in a 105.0 ns window starting 3.0 ns before (F)ADC data.
Camera filter 1-D transmission table applied for telescope 3.
Average night sky background of telescope 3: 0.218 p.e./ns (DC current: 1.40 uA)

Start setting up telescope 4.
Preprocessor command: /workdir/sim_telarray/sim_telarray/bin/pfp -v -I. -DNUM_TELESCOPES=1 -DWITH_LOW_GAIN_CHANNEL -DMAX_GAINS=2 -DWITH_GSL_RNG -DSIMTEL_VERSION=1709915244 -DSIMTEL_RELEASE=20240780 -I../cfg/CTAiobuf_maximum=1000000000 -I. -I/workdir/sim_telarray/sim_telarray/cfg -I/workdir/sim_telarray/sim_telarray/cfg/common -I/workdir/sim_telarray/sim_telarray/cfg/hess -I/workdir/sim_telarray/sim_telarray/cfg/hess2 -I/workdir/sim_telarray/sim_telarray/cfg/hess3 -I/workdir/sim_telarray/sim_telarray/cfg/hess5000 -I/workdir/sim_telarray/sim_telarray/cfg/CTA -I/workdir/sim_telarray/sim_telarray/cfg/MAGIC -DMAX_GAINS=2 -DTELESCOPE=4 - < /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg
Applying 0 weak command-line configuration parameter(s).
Check additional configuration for telescope 4 from /workdir/external/simtools/simtools-output/simulate_light_emission/model/CTA-North-MSTN-04-6.0.0_simulate_light_emission.cfg.
Configuration for MSTN-04 - TELESCOPE 4
137 configuration lines were processed.
Applying 10 forced command-line configuration parameter(s).
Filled 8 metaparameters for telescope ID 4 (itel=3).
Camera 4 by default requires 3 pixels per trigger group.
Reading setup for telescope 4 camera from file camera_CTA-MST-NectarCam_20191120_majority-3nn.dat
Pixel type 1 has a hexagonal (flat y) funnel onto a circular visible cathode.
Reading funnel angle efficiencies for telescope 4 from NectarCAM_lightguide_efficiency_POP_131019.dat
Table with 47 rows has been read from file NectarCAM_lightguide_efficiency_POP_131019.dat
Telescope 4 has 86 mirrors (from file mirror_CTA-100_1.20-86-0.04.dat).
Geometric area of 86 mirrors (inclination corrected): 106.26 m^2.
Diameter is approximately 13.775 m
Effective focal length given as 1644.5 for nominal 1600.0 cm.
Random reflections with 0.0068 deg.
Warning: trigger window does not cover full readout window.
(F)ADC data is simulated for a readout window of up to 105.0 ns,
trigger conditions in a 105.0 ns window starting 3.0 ns before (F)ADC data.
Camera filter 1-D transmission table applied for telescope 4.
Average night sky background of telescope 4: 0.218 p.e./ns (DC current: 1.40 uA)



```

**Comment by orelgueta** [comment]
I would also suggest avoiding using the `pfp` preprocessor and reduce the default include paths used to just the ones necessary (i.e., not HESS/MAGIC ones). We should try to decouple as much as possible from the built in scripts.

**Comment by tobiaskleiner** [comment]
> I would also suggest avoiding using the `pfp` preprocessor and reduce the default include paths used to just the ones necessary (i.e., not HESS/MAGIC ones). We should try to decouple as much as possible from the built in scripts.

This would need a change in sim_telarray. I have added an issue https://github.com/gammasim/simtools/issues/1132

File: .gitignore
Comment (GernotMaier) [inline .gitignore:1]: Please remove those changes. They seem to be more related to your local environment?

File: simtools/simtel/simulator_light_emission.py [hunk 1]
```diff
@@ -398,6 +354,26 @@ def remove_line_from_config(file_path, line_prefix):
 
         return command
 
+    def _remove_line_from_config(self, file_path, line_prefix):
```
Comment (GernotMaier) [inline simtools/simtel/simulator_light_emission.py:357]: Do you have a unit test for this function? I didn't see it.

**Review by GernotMaier — CHANGES_REQUESTED** [review]
See comments below. I think the missing unit test is also the reason why sonarq is failing.

File: simtools/simtel/simulator_light_emission.py (hunk 1 above)
Comment (tobiaskleiner) [reply simtools/simtel/simulator_light_emission.py:357]: Thanks for pointing that out. A test is included now.

File: .gitignore
Comment (tobiaskleiner) [reply .gitignore:1]: I have noticed these files are sometimes created from the light emission package when interrupted before finishing.
Not sure how the .swp file has been created, but I can remove it if it is not appearing for you.

**Review by GernotMaier — APPROVED** [review]
Great, thank you!

**Pull request merged**